File: Python3/2423_remove_letter_to_equalize_frequency.py
# https://leetcode.cn/problems/remove-letter-to-equalize-frequency/

class Solution:
    def equalFrequency(self, word):
        """
        Inputs:
            word: str
        Outputs:
            res: bool
        """
        cnt_tmp = [0 for _ in range(26)]
        for ch in word:
            cnt_tmp[ord(ch)-ord("a")] += 1
        cnt = []
        for c in cnt_tmp:
            if c:
                cnt.append(c)
        for i in range(len(cnt)):
            if cnt[i] == 1:
                tmp = cnt[:i] + cnt[i+1:]
            else:
                tmp = cnt[:i] + [cnt[i]-1] + cnt[i+1:]
            if len(set(tmp)) == 1:
                return True
        return False

File: Cpp/2423_remove_letter_to_equalize_frequency.cpp
// https://leetcode.cn/problems/remove-letter-to-equalize-frequency/

class Solution {
public:
    bool equalFrequency(string word) {
        vector<int> cnt_tmp(26, 0);
        for (char ch: word) {
            cnt_tmp[ch - 'a'] += 1;
        }
        vector<int> cnt;
        for (int c: cnt_tmp) {
            if (c > 0) cnt.push_back(c);
        }
        for (int i = 0; i < cnt.size(); ++i) {
            cnt[i] -= 1;
            unordered_set<int> s;
            for (int j = 0; j < cnt.size(); ++j) {
                if (cnt[j] > 0) {
                    s.insert(cnt[j]);
                }
            }
            if (s.size() == 1) return true;
            cnt[i] += 1;
        }
        return false;
    }
};
